internal/configsource/etcd2configsource/mocks_test.go

```diff
@@ -18,23 +18,24 @@ package etcd2configsource
 import (
 	"context"
 	"errors"
+	"sync/atomic"
```

Comment:
Can we add uber atomic to the allowed list instead of replacing it with go atomic?
There was a long discussion around this in upstream and the decision came in favor of uber's package
https://github.com/open-telemetry/opentelemetry-collector/issues/5200#issuecomment-1097884180

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/issues/7160

Reply:
Thanks! in this case, LGTM